tests/test_cli.py

```diff
@@ -182,3 +191,169 @@ def test_find_designated_time_function(self):
             answer_3, answer_4 = cli_download.find_designated_time(True, '500', '')
             self.assertEqual(answer_3, datetime(2011, 12, 24, 4, 0, tzinfo=tzutc()))
             self.assertEqual(answer_4, datetime(2012, 1, 14, 0, 0, tzinfo=tzutc()))
+
+
+class CopyingCLITestCase(django.test.TestCase):
+    """Tests for the cli of copying """
+
+    fixtures = [os.path.join(os.path.dirname(__file__), 'data/test_data.json')]
+
+    def test_extract_arg(self):
+        """shall return the correct argument values based on the 'sys.argv' """
+        sys.argv = [
+            "",
+            '-d', "/test_folder/",
+            '-b', "200",
+            '-e', "2018-11-18",
+            '-r',
+            '-f',
+            '-l',
+            '-g', "POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))",
+            '-t', 'test_type',
+            '-q',
+            '{"dataseturi__uri__contains": "osisaf", "source__instrument__short_name__icontains": '
+            + '"AMSR2"}'
+        ]
+        arg = cli_copy.cli_parse_args()
+        self.assertEqual(arg.begin, '200')
+        self.assertEqual(arg.destination_path, '/test_folder/')
+        self.assertEqual(arg.end, '2018-11-18')
+        self.assertEqual(arg.geometry, 'POLYGON ((-22 84, -22 74, 32 74, 32 84, -22 84))')
+        self.assertEqual(arg.type, 'test_type')
+        self.assertEqual(arg.query,
+                         '{"dataseturi__uri__contains": "osisaf", '
+                         + '"source__instrument__short_name__icontains": "AMSR2"}')
+        # testing the flag enumeration
+        self.assertTrue(arg.rel_time_flag)
+        self.assertTrue(arg.flag_file)
+        self.assertTrue(arg.link)
+        sys.argv.remove('-r')
+        sys.argv.remove('-f')
+        sys.argv.remove('-l')
+        arg = cli_copy.cli_parse_args()
+        self.assertFalse(arg.rel_time_flag)
+        self.assertFalse(arg.flag_file)
+        self.assertFalse(arg.link)
```

Comment:
Because first we have to assert the flag is conveyed corrected to this step of workflow. And then by removing the flag once again make ourselves secured of falseness of this values. So two aspects must be tested apart from each other.

Reply:
Then why don't you test the options which take an argument in the same way?

Reply:
What do you mean? these are flags are change the behaviour by their presence and absence. The others are tested by their values! Is there any more tests needed?

Reply:
What I mean is: what happens if the others are absent?

Reply:
Others are not working by their absence or presence. They are working by their values, which are tested in the above lines! These are the one that are resulting in different situation by their absence and presence and both `absence` and `presence` situation of them must be approved to be false or true.